edalize/libero.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import os.path
+import os
+import platform
+import subprocess
+import re
+import xml.etree.ElementTree as ET
+from functools import partial
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+
+class Libero(Edatool):
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            return {'description': "The Libero backend supports Microsemi Libero to build systems and program the FPGA",
+                    'members': [
+                        {'name': 'family',
+                         'type': 'String',
+                         'desc': 'FPGA family (e.g. PolarFire)'},
+                        {'name': 'die',
+                         'type': 'String',
+                         'desc': 'FPGA device (e.g. MPF300TS)'},
+                        {'name': 'package',
+                         'type': 'String',
+                         'desc': 'FPGA package type (e.g. FCG1152)'},
+                        {'name': 'speed',
+                         'type': 'String',
+                         'desc': 'FPGA speed rating (e.g. -1)'},
+                        {'name': 'dievoltage',
+                         'type': 'String',
+                         'desc': 'FPGA die voltage (e.g. 1.0)'},
+                        {'name': 'range',
+                         'type': 'String',
+                         'desc': 'FPGA temperature range (e.g. IND)'},
+                        {'name': 'defiostd',
+                         'type': 'String',
+                         'desc': 'FPGA default IO std (e.g. "LVCMOS 1.8V"'},
+                        {'name': 'hdl',
+                         'type': 'String',
+                         'desc': 'Default HDL (e.g. "VERILOG"'}
+                    ]
+                    }
+
+    argtypes = ['vlogdefine', 'vlogparam']
+
+    tool_options_defaults = {
+        'speed': '-1',
+        'dievoltage': '1.0',
+        'range': 'IND',
+        'defiostd': 'LVCMOS 1.8V',
+        'hdl': 'VERILOG',
+    }
```

Comment:
I'm always a bit sceptical when it comes to setting default values. Are these the most common values? Do all devices support dievoltage 1.0 etc. Unless we can confirm these are reasonable defaults I think it's better to leave them completely to the user and not write them to the template unless they have been set by the user

Reply:
Yes, let's remove all of them. It's better that more experience Libero users tell us later which defaults that make sense

Reply:
See next comment.